aes-kw/src/kek.rs

```diff
@@ -0,0 +1,196 @@
+use aes::cipher::{BlockDecrypt, BlockEncrypt, NewBlockCipher};
+use aes::BlockCipher;
+use generic_array::typenum::{U16, U24, U32};
+use generic_array::GenericArray;
+
+const IV_LEN: usize = 8;
+const IV: [u8; IV_LEN] = [0xA6, 0xA6, 0xA6, 0xA6, 0xA6, 0xA6, 0xA6, 0xA6];
+
+use crate::Error;
+
+/// A KEK that can be used to wrap and unwrap.
+#[derive(Debug, Clone, Copy, PartialEq)]
+pub struct Kek<Aes>
+where
+    Aes: NewBlockCipher + BlockCipher<BlockSize = U16> + BlockEncrypt + BlockDecrypt,
+{
+    /// Initialized cipher
+    cipher: Aes,
+}
+
+impl From<GenericArray<u8, U16>> for Kek<aes::Aes128> {
+    fn from(kek: GenericArray<u8, U16>) -> Self {
+        Kek::new(&kek)
+    }
+}
+
+impl From<GenericArray<u8, U24>> for Kek<aes::Aes192> {
+    fn from(kek: GenericArray<u8, U24>) -> Self {
+        Kek::new(&kek)
+    }
+}
+
+impl From<GenericArray<u8, U32>> for Kek<aes::Aes256> {
+    fn from(kek: GenericArray<u8, U32>) -> Self {
+        Kek::new(&kek)
+    }
+}
+
+impl From<[u8; 16]> for Kek<aes::Aes128> {
+    fn from(kek: [u8; 16]) -> Self {
+        Kek::new(&kek.into())
+    }
+}
+
+impl From<[u8; 24]> for Kek<aes::Aes192> {
+    fn from(kek: [u8; 24]) -> Self {
+        Kek::new(&kek.into())
+    }
+}
+
+impl From<[u8; 32]> for Kek<aes::Aes256> {
+    fn from(kek: [u8; 32]) -> Self {
+        Kek::new(&kek.into())
+    }
+}
+
+impl<Aes> std::convert::TryFrom<&[u8]> for Kek<Aes>
+where
+    Aes: NewBlockCipher + BlockCipher<BlockSize = U16> + BlockEncrypt + BlockDecrypt,
+{
+    type Error = Error;
+
+    fn try_from(value: &[u8]) -> Result<Self, Self::Error> {
+        match value.len() * 8 {
+            128 => Ok(Kek::new(GenericArray::from_slice(value))),
+            192 => Ok(Kek::new(GenericArray::from_slice(value))),
+            256 => Ok(Kek::new(GenericArray::from_slice(value))),
+            v => Err(Error::InvalidKekSize(v)),
+        }
+    }
+}
+
+impl<Aes> std::convert::TryFrom<Vec<u8>> for Kek<Aes>
+where
+    Aes: NewBlockCipher + BlockCipher<BlockSize = U16> + BlockEncrypt + BlockDecrypt,
+{
+    type Error = Error;
+
+    fn try_from(value: Vec<u8>) -> Result<Self, Self::Error> {
+        match value.len() * 8 {
+            128 => Ok(Kek::new(GenericArray::from_slice(&value))),
+            192 => Ok(Kek::new(GenericArray::from_slice(&value))),
+            256 => Ok(Kek::new(GenericArray::from_slice(&value))),
+            v => Err(Error::InvalidKekSize(v)),
+        }
```

Comment:
```suggestion
        value.as_slice().try_into()
```

...also it seems like that's easy enough, maybe this impl is unnecessary?

Reply:
you are right, removed it